domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/caserole/CaseRoleEntityFieldValueValidatorImpl.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.hmcts.ccd.definition.store.domain.validation.caserole;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.SimpleValidationError;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.ValidationErrorMessageCreator;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.ValidationResult;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.CaseRoleEntity;
+
+@Component
+public class CaseRoleEntityFieldValueValidatorImpl implements CaseRoleEntityValidator {
+
+    private static final String CASE_ROLE_ID_REGEX = "^(\\[[A-Za-z]+\\])$";
+    private static final int ID_MAX_LENGTH = 40;
+    private static final int NAME_MAX_LENGTH = 40;
+
+    @Override
+    public ValidationResult validate(CaseRoleEntity caseRoleEntity,
+                                     CaseRoleEntityValidationContext caseRoleEntityValidationContext) {
+        final ValidationResult validationResult = new ValidationResult();
+        if (caseRoleEntity.getReference() != null &&
+            !caseRoleEntity.getReference().matches(CASE_ROLE_ID_REGEX)) {
```

Comment:
story mentions:
`ids should always be stored as uppercase`.
Here we allow id with lowercase. Just to double check, are we converting all to uppercase before saving?